internal/provider/waypoint/resource_waypoint_application.go

```diff
@@ -146,6 +212,38 @@ func (r *ApplicationResource) Configure(ctx context.Context, req resource.Config
 	r.client = client
 }
 
+type varConverter struct {
+	name         string
+	variableType string
+	value        string
+}
+
+// FromTerraform5Value implements the ValueConverter interface
+func (vc *varConverter) FromTerraform5Value(val tftypes.Value) error {
```

Comment:
```suggestion
func (vc *varConverter) FromTerraformValue(val tftypes.Value) error {
```
Is this a typo?

Reply:
`FromTerraform5Value` is correct, it implements the [`ValueConverter` interface](https://github.com/hashicorp/terraform-plugin-go/blob/05dc75aefa5b71406022d0ac08eca99f44fbf378/tftypes/value.go#L17-L24), allowing us to use `As` [here](https://github.com/hashicorp/terraform-provider-hcp/pull/833/files#diff-523041116371f0db18a6aa9f5699c5686d44f0bfea408dd44545e0ee5d868a5aR295).

Reply:
FYI @HenryEstberg I ultimately ended up removing this code after chatting with @catsby and pivoting to using `ElementsAs` to do the conversion, over implementing the conversion in the loop later in this file via `As` (which is what used this interface implementation).